packages/flutter_tools/lib/src/base/command_help.dart

```diff
@@ -0,0 +1,71 @@
+
+  factory CommandHelp.L() => CommandHelp._('L', 'Dump layer tree to the console.', 'debugDumpLayerTree');
+  factory CommandHelp.P() => CommandHelp._('P', 'Toggle performance overlay.', 'WidgetsApp.showPerformanceOverlay');
+  factory CommandHelp.R() => CommandHelp._('R', 'Hot restart');
+  factory CommandHelp.S() => CommandHelp._('S', 'Dump accessibility tree in traversal order.', 'debugDumpSemantics');
+  factory CommandHelp.U() => CommandHelp._('U', 'Dump accessibility tree in inverse hit test order.', 'debugDumpSemantics');
+  factory CommandHelp.a() => CommandHelp._('a', 'Toggle timeline events for all widget build methods.', 'debugProfileWidgetBuilds');
+  factory CommandHelp.d() => CommandHelp._('d', 'Detach (terminate "flutter run" but leave application running).');
+  factory CommandHelp.h() => CommandHelp._('h', 'Repeat this help message.');
+  factory CommandHelp.i() => CommandHelp._('i', 'Toggle widget inspector.', 'WidgetsApp.showWidgetInspectorOverride');
+  factory CommandHelp.o() => CommandHelp._('o', 'Simulate different operating systems.', 'defaultTargetPlatform');
+  factory CommandHelp.p() => CommandHelp._('p', 'Toggle the display of construction lines.', 'debugPaintSizeEnabled');
+  factory CommandHelp.q() => CommandHelp._('q', 'Quit (terminate the application on the device).');
+  factory CommandHelp.r() => CommandHelp._('r', 'Hot reload. $fire$fire$fire');
+  factory CommandHelp.s() => CommandHelp._('s', 'Save a screenshot to flutter.png.');
+  factory CommandHelp.t() => CommandHelp._('t', 'Dump rendering tree to the console.', 'debugDumpRenderTree');
+  factory CommandHelp.w() => CommandHelp._('w', 'Dump widget hierarchy to the console.', 'debugDumpApp');
+  factory CommandHelp.z() => CommandHelp._('z', 'Toggle elevation checker.');
+
+  /// The key associated with this command
+  final String key;
+  /// A description of what this command does
+  final String description;
+  /// Text shown in parenthesis to give the context
+  final String inParenthesis;
+
+  bool get _hasTextInParenthesis => inParenthesis != null && inParenthesis.isNotEmpty;
+
+  @override
+  String toString() {
+    final StringBuffer message = StringBuffer();
+    message.writeAll(<String>[terminal.bolden(key), description], ' ');
+
+    if (_hasTextInParenthesis) {
+      bool wrap = false;
+      final int maxWidth = max(io.stdio.terminalColumns ?? 0,  maxLineWidth);
+      int width = maxWidth - message.length;
+      final String parentheticalText = '($inParenthesis)';
+      if (width < parentheticalText.length) {
+        width = maxWidth;
+        wrap = true;
+      }
+      message.writeAll(
+          <String>[
+            if (wrap) '\n',
+            terminal.color(
+              parentheticalText,
+              TerminalColor.grey,
+            ).padLeft(width)
+          ]);
+    }
+    return message.toString();
+  }
+
+  void print() {
+    printStatus(toString());
+  }
```

Comment:
I think rather than `CommandHelp.w().print();` we should go with `printStatus(CommandHelp.w());`, it would be more idiomatic.

Reply:
(and makes this function unnecessary)

Reply:
That was my initial goal but `printStatus` takes a `String` argument and we would have to do `printStatus(CommandHelp.w().toString())` which is a little bit verbose, that's why I encapsulated printing.

If I break too much the idiomatic uniformity of the project, I can change it.

Reply:
Hm, I see what you're saying. I agree that that wouldn't be great. I can go with the way you did it, sure.